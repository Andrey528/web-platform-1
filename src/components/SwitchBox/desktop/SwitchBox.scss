.switch-box {
  width: 2.5rem;
  height: 1rem;
  border-radius: 0.5rem;
  background: #6C7883;
  display: flex;
  align-items: center;
  cursor: pointer;
  transition: all .2s ease-in;
  &_active {
    background: #5288C1;
    .switch-box__indicator {
      border-color: #5288C1;
      transform: translateX(1.2rem);
    }
  }
  &__indicator {
    width: 1rem;
    height: 1rem;
    border: 0.2rem solid #6C7883;
    border-radius: 100%;
    background: #182533;
    transition: all .2s ease-in;
  }
}