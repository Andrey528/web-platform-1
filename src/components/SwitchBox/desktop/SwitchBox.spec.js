import React from 'react';
import { shallow } from 'enzyme';
import block from 'bem-cn';

import { SwitchBox } from './SwitchBox';

describe('SwitchBox tests', () => {
  const b = block('switch-box');
  const props = {
    isActive: true,
    callBack: jest.fn(),
  };
  const className = b({ active: props.isActive });

  describe('SwitchBox with props', () => {
    const switchBox = shallow(<SwitchBox {...props} />);

    it('render properly', () => {
      expect(switchBox).toMatchSnapshot();
    });
  });

  describe('SwitchBox have modificators', () => {
    it('Have modificators', () => {
      const switchBox = shallow(<SwitchBox {...props} />);

      expect(switchBox.find(`div.${b().toString()}`).props().className.toString()).toEqual(className.toString());
    });
  });

  describe('Callback test', () => {
    it('onClick working', () => {
      const switchBox = shallow(<SwitchBox {...props} />);
      switchBox.find(`div.${b().toString()}`).simulate('click');
      expect(props.callBack).toHaveBeenCalledTimes(1);
    });

    it('onClick multiply times', () => {
      const switchBox = shallow(<SwitchBox {...props} />);

      switchBox.find(`div.${b().toString()}`).simulate('click');
      switchBox.find(`div.${b().toString()}`).simulate('click');
      switchBox.find(`div.${b().toString()}`).simulate('click');

      expect(props.callBack).toHaveBeenCalledTimes(3);
    });
  });
});