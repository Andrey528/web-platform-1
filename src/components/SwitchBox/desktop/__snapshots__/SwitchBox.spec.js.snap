// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchBox tests SwitchBox with props render properly 1`] = `
<div
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  onClick={[Function]}
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  />
</div>
`;
